File: Week1_Design Patterns and Principles/Ex_2_Implementing_the_Factory_Method_Pattern/Code/com/example/factory/factories/ExcelDocumentFactory.java
package com.example.factory.factories;
import com.example.factory.documents.Document;
import com.example.factory.documents.ExcelDocument;
public class ExcelDocumentFactory extends DocumentFactory {
    @Override
    public Document createDocument() {
        System.out.println("ExcelDocumentFactory: Creating new Excel document...");
        return new ExcelDocument();
    }
}

File: Week1_Design Patterns and Principles/Ex_2_Implementing_the_Factory_Method_Pattern/Code/Factories/ExcelDocumentFactory.cs
using System;
using FactoryDemo.Documents;

namespace FactoryDemo.Factories;

public sealed class ExcelDocumentFactory : DocumentFactory
{
    public override IDocument CreateDocument()
    {
        Console.WriteLine("ExcelDocumentFactory: Creating new Excel document…");
        return new ExcelDocument();
    }
}
